src/Storages/MergeTree/MergeTreePrefetchedReadPool.cpp

```diff
@@ -419,10 +401,6 @@ MergeTreePrefetchedReadPool::ThreadsTasks MergeTreePrefetchedReadPool::createThr
     }
 
     size_t min_prefetch_step_marks = 0;
-    if (settings.filesystem_prefetches_limit && settings.filesystem_prefetches_limit < sum_marks)
```

Comment:
this is to actually have moderate const-size tasks. and when we will have code to start new prefetch as soon as some of the previous completed, all tasks will be prefetched as soon as `max_streams` tasks fit in limits.